<launch>

  <env name="LIBGL_ALWAYS_SOFTWARE" value="1"/>

  <arg name="config" default="crg_segmentation" />

  <!-- read value of arg -->
  <param name="param" value="$(arg config)"/> 
  
  <rosparam file="$(find seam_detection)/config/$(arg config).yaml" />

  <!-- setup the transformations and frames for the system-->
  <node
    pkg="seam_detection"
    name="filter_cloud_tf"
    type="filter_cloud_tf"
    output="screen"
  />                      
  
  <node name="rviz" 
    pkg="rviz" 
    type="rviz" 
    args="-d $(find seam_detection)/rviz/crg_segmentation.rviz">
  </node>

  <node                     
    pkg="seam_detection"          
    name="crg_segmentation"   
    type="crg_segmentation"
    output="screen"  
   />
   
</launch>
